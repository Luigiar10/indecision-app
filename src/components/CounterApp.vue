<template>
  <h2>{{ checkTitle }}</h2>
  <p>{{counter}} <sup>2</sup> = {{ mathForm }}</p>
  <button class="btn" @click="plusNumber()">Sumar</button>
  <button class="btn" v-on:click="minusNumber()">Restar</button>
</template>

<script>
export default {
    props: {
        title: String,
        start: {
            type: Number,
            default: 10,
            // required: true
            validator( value ) {
                return value >= 10
            }
        }
    },
    name: 'CoounterApp',

    data() {
        return {
            counter: this.start
        }
    },
    
    methods: {
        plusNumber() {
            this.counter = this.counter + 1
        },

        minusNumber() {
            this.counter = this.counter - 1
        }
    },

    // Las propiedades computadas tienen que regresar un valor
    computed: {
        mathForm() {
            return this.counter * this.counter
        },

        checkTitle() {
            // return this.title ? this.title : 'Counter'
            return this.title || 'Counter'
        }
    }
}
</script>

<style>
    .btn {
        background-color: #64bb87;
        border-radius: 10px;
        border: none;
        cursor: pointer;
        margin: 2rem;
        padding: .4rem 1.4rem;
        transition: 0.3s ease-in-out;
    }

    .btn:hover {
        background-color: #5aa67b;
        transition: 0.3s ease-in-out;
    }
</style>