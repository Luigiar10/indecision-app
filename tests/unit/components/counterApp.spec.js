import {shallowMount} from '@vue/test-utils'
import CounterApp from '@/components/CounterApp'

describe('Counter App Test', () => {
    test('Debe de hacer match con el snapshot', () => {
        const wrapper = shallowMount(CounterApp)
        expect( wrapper.html() ).toMatchSnapshot()
    })
})