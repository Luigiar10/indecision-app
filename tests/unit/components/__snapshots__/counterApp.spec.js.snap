// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter App Test Debe de hacer match con el snapshot 1`] = `
<h2>Counter</h2>
<p>10 <sup>2</sup> = 100</p><button class="btn btn-danger">Sumar</button><button class="btn">Restar</button>
`;
